cura/Machines/ContainerTree.py

```diff
@@ -76,54 +74,82 @@ def getCurrentQualityChangesGroups(self) -> List["QualityChangesGroup"]:
-        Logger.log("d", "Adding container tree for {definition_id} took {duration} seconds.".format(definition_id = definition_id, duration = time.time() - start_time))
-
-    ##  Builds the initial container tree.
-    def _loadAll(self):
-        Logger.log("i", "Building container tree.")
-        start_time = time.time()
-        all_stacks = ContainerRegistry.getInstance().findContainerStacks()
-        for stack in all_stacks:
-            if not isinstance(stack, GlobalStack):
-                continue  # Only want to load global stacks. We don't need to create a tree for extruder definitions.
-            definition_id = stack.definition.getId()
-            if definition_id not in self.machines:
-                definition_start_time = time.time()
-                self.machines[definition_id] = MachineNode(definition_id)
-                self.machines[definition_id].materialsChanged.connect(self.materialsChanged)
-                Logger.log("d", "Adding container tree for {definition_id} took {duration} seconds.".format(definition_id = definition_id, duration = time.time() - definition_start_time))
-
-        Logger.log("d", "Building the container tree took %s seconds",  time.time() - start_time)
-
-    ##  When a printer gets added, we need to build up the tree for that container.
-    def _machineAdded(self, container_stack: ContainerInterface) -> None:
-        if not isinstance(container_stack, GlobalStack):
-            return  # Not our concern.
-        self.addMachineNodeByDefinitionId(container_stack.definition.getId())
-
-    ##  For debugging purposes, visualise the entire container tree as it stands
-    #   now.
-    def _visualise_tree(self) -> str:
-        lines = ["% CONTAINER TREE"]  # Start with array and then combine into string, for performance.
-        for machine in self.machines.values():
-            lines.append("  # " + machine.container_id)
-            for variant in machine.variants.values():
-                lines.append("    * " + variant.container_id)
-                for material in variant.materials.values():
-                    lines.append("      + " + material.container_id)
-                    for quality in material.qualities.values():
-                        lines.append("        - " + quality.container_id)
-                        for intent in quality.intents.values():
-                            lines.append("          . " + intent.container_id)
-        return "\n".join(lines)
\ No newline at end of file
+    ##  Ran after completely starting up the application.
+    def _onStartupFinished(self):
+        currently_added = ContainerRegistry.getInstance().findContainerStacks()  # Find all currently added global stacks.
+        JobQueue.getInstance().add(self.MachineNodeLoadJob(self, currently_added))
+
+    ##  Dictionary-like object that contains the machines.
+    #
+    #   This handles the lazy loading of MachineNodes.
+    class MachineNodeMap:
```

Comment:
Since this is a private class (right?) it should be _MachineNodeMap. If it's not a private class it should be in its own file.

Reply:
Done: 76f54256230459c4df2a7977a5a2fff4b693b638